clang/lib/CodeGen/Targets/X86.cpp

```diff
@@ -1069,6 +1069,10 @@ Address X86_32ABIInfo::EmitVAArg(CodeGenFunction &CGF,
 
   auto TypeInfo = getContext().getTypeInfoInChars(Ty);
 
+  // Empty records are ignored for parameter passing purposes on non-Windows.
+  if (!IsWin32StructABI && isEmptyRecord(getContext(), Ty, true))
+    return CGF.CreateMemTemp(Ty);
```

Comment:
I'd prefer to use classifyArgumentType, like we do for the 64-bit case, instead of trying to duplicate the logic.

Reply:
If it doesn't work with the code as-is, please refactor to the code to make it work.  Scattered checks like this are hard to maintain.

The simplest thing here would be to just construct a CCState here, I think?  Or if that doesn't work for some reason, make the CCState parameter to classifyArgumentType optional.